pyscriptjs/src/pyodide.ts

```diff
@@ -80,8 +81,28 @@ export class PyodideInterpreter extends Interpreter {
         logger.info('pyodide loaded and initialized');
     }
 
-    run(code: string): unknown {
-        return this.interface.runPython(code);
+    /**
+     * Run the given code in the interpreter, optionally you can pass an options
+     * object which will be added to the interpreter's globals. Ideally we would
+     * pass the options to `locals` but Pyodide's `runPython` doesn't support
+     * that yet.
+     *
+     * @param code - Python code to run in the interpreter
+     * @param options? - Options to pass to the interpreter (currently, only used for `globals`)
+     * @returns The result of the code execution
+     */
+    run(code: string, options?: RunOptions): unknown {
+        // Since Pyodide doesn't support passing `locals` to `runPython` we
+        // just iterate over the globals. Once(if) it does, we should move
+        // this logic to `locals` or extend to support both.
+        const additionalGlobals = options?.additionalGlobals;
+        if (additionalGlobals && Object.keys(additionalGlobals).length > 0) {
+            for (const [key, value] of Object.entries(additionalGlobals)) {
+                this.globals.set(key, value);
+            }
+        }
+
+        return this.interpreter.runPython(code);
```

Comment:
Why not `pyodide.pyodide_py.code.eval_code.callKwargs(code, {locals: someLocalsDict});`?